File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.java
class Solution {
    public int subarraySum(int[] nums, int k) {
        HashMap<Integer,Integer> map= new HashMap<>();
        int cnt=0;
        int sum=0;
        for(int i=0;i<nums.length;i++){
            sum+=nums[i];
            if(sum==k){
                cnt++;
            } if(map.containsKey(sum-k)){
                cnt+=map.get(sum-k);
            }
            map.put(sum, map.getOrDefault(sum, 0) + 1);
        }
      return cnt;  
    }
}

File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
    unordered_map<int, int> map;
    int n = nums.size();
    int cnt = 0;
    int sum = 0;

    for (int i = 0; i < n; i++) {
        sum += nums[i];
        if (sum == k) {
            cnt++;
        }
        if (map.find(sum - k) != map.end()) {
            cnt += map[sum - k];
        }
        map[sum]++;
    }

    return cnt;
}

    
};